folium/map.py

```diff
@@ -243,13 +253,17 @@ class Marker(MacroElement):
         {% endmacro %}
         """)
 
-    def __init__(self, location, popup=None, tooltip=None, icon=None):
+    def __init__(self, location=None, popup=None, tooltip=None, icon=None,
+                 **kwargs):
         super(Marker, self).__init__()
-        self._name = 'Marker'
-        self.tooltip = tooltip
-        self.location = _validate_coordinates(location)
+        self._name = 'marker'
+        if location is not None:
```

Comment:
I get that you want to do this for using it in `GeoJson`. I'm worried though that people forget to provide coordinates, and then don't get an error but a blank map instead, since the error is happening in the browser. What about leaving the `location` argument without default value? So no `location=None`, but still accept that `location` may be `None` when used in `GeoJson`.

Reply:
Is there some way to check the type of the parent class the Marker is added to, and validate there? I guess the problem is there are so many different classes that these can be children of - `FeatureGroup`, `MarkerCluster`, directly to `Map`, etc. Would a `render` function, like many other classes have be appropriate? We could call it in any template where a marker is added to a child, and validate there.

It would be nice if the user did not have to explicitly pass `location = None` because it's not really intuitive at face value - it's not `None`, it just has many possible values instead of one. I get we can state it in the `GeoJson` docstring, but I'd like it to be clean and straightforward for users.

Reply:
> Would a render function, like many other classes have be appropriate?

Sounds good. Add a `render` method to `Marker` that checks if `self.location` is not None. In your change the `marker` in `GeoJson` is not added as a child, so we don't even have to do any logic, since `marker` is not in the rendering tree.

Can you add this? If we do this at `Marker`, we got them all right? Since all other marker classes you set `location=None` inherit from `Marker` and don't have their own `render` method.